raiden/network/transport/matrix.py

```diff
@@ -705,43 +723,35 @@ def _handle_invite(self, room_id: _RoomID, state: dict):
             join_rules_event = join_rules_events[0]
             private_room: bool = join_rules_event['content'].get('join_rule') == 'invite'
 
-        # join room async, so it doesn't block handling loop
-        def join_room():
```

Comment:
This is blocking the calling thread again, I'm curious, which thread one is it? the sync thread?